proto/redpanda/core/admin/v2/shadow_link.proto

```diff
@@ -535,6 +535,8 @@ message ShadowLinkTaskStatus {
     string reason = 3;
     // The broker the task is running on
     int32 broker_id = 4;
+    // The shard the task is running on
+    int32 shard = 5;
```

Comment:
Probably fine, but we have talked about in the past not exposing shard level information in cloud, but I get that this is useful to know.